### DISCUSSION TITLE
How to make filter with simple checkbox using a relationship

### DISCUSSION THREAD

**Discussion by slossetti** [rappasoft/laravel-livewire-tables#1347] [Q&A] [answered]
Hi,

I need to make a filer with only a checkbox. I want to user the ability to filter by favorites, and I have the following function into my Model, but I want to know how can add a filter for that.

`    public function favorites()
    {
        // Obtén el usuario logueado
        $loggedInUser = auth()->user();

        if ($loggedInUser) {
            return $this->hasMany(Favorite::class)->where('user_id', $loggedInUser->id);
        }
    }`
    
    Like this:
    
    
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/57014758/63028d5b-5b7b-4ba2-b12a-9075f72e5b93)

Thanks!

**Comment by lrljoe** [comment, marked as answer]
You can use a Select Filter for this.  You can pass a blank blade to the setCustomFilterLabel() method and it won't have a label anymore (so more what you showed)

Alternatively, you could use the configurable areas, entangle it with a public variable, and use that in your builder() method.

[https://rappasoft.com/docs/laravel-livewire-tables/v2/datatable/configurable-areas](https://rappasoft.com/docs/laravel-livewire-tables/v2/datatable/configurable-areas)

A toggle filter is in the works, and it won't be difficult to add a single checkbox option too.  It won't be out until later this month tho
